### PR TITLE
Better handling of unauthorizedInterrupt, keyName, $cancel tweaks

### PR DESCRIPTION
**Pull request by Tathanen**
New features:

* `APIRequestUnauthorized` is broadcast if requests repeated after a 401 resolution still return a 401
* `$cancel` now halts the HTTP request itself, and doesn't reject the main promise by default
* `keyName` method allows customization of the authorization header name

Additional breaking changes/notes:

* If a repeated request after a 401 resolution still returns a 401, any other requests in the repeat queue are silently canceled
* `$cancel` no longer rejects the main promise by default, but can be instructed to do so by passing the `reject` parameter
* The authorization header has been renamed from `AUTHORIZATION` to `Authorization`.  Since HTTP headers are supposed to be case-insensitive this _shouldn't_ be breaking, but in case you were referencing it explicitly in case-sensitive JavaScript, you can use `keyName` to change it back.

I'm using this in a project currently to make sure it's all up to snuff, I'll merge/release it once I'm sure it's solid.  In the meantime, please review @vokal/web

### CODE REVIEW

File: dist/vokal-ng-api.js [hunk 1]
```diff
@@ -341,10 +355,31 @@ angular.module( "vokal.API", [ "vokal.Humps" ] )
 
                                 return;
                             }
+                            else
+                            {
+                                $rootScope.$broadcast( "APIRequestUnauthorized", options, data, status );
```
Comment (WorkHorseIndustries) [inline dist/vokal-ng-api.js]: don't need to return here? 
If this is the same as `that.unauthorizedInterrupt === true` you could do a check for 
```javascript
if ( typeof that.unauthorizedInterrupt === "function" )
{ 
...
}
else 
{
...
}`

**Comment by WorkHorseIndustries** [comment]
FWIW LGTM

File: dist/vokal-ng-api.js (hunk 1 above)
Comment (Tathanen) [inline dist/vokal-ng-api.js]: Yeah, `=== true` means you want your normal error handling to be interrupted, thus the return.  Same with when you provide a function.  If you don't set it to true, it still broadcasts the unauthorized event, but the lack of a `return` lets it also continue to the promise rejection block later on.

**Comment by jrit** [comment]
`If a repeated request after a 401 resolution still returns a 401, any other requests in the repeat queue are silently canceled`

meaning really silent, like absolutely nothing happens, or it broadcasts an event like `requestQueueFlushed`? I think having something observable would be a good idea, if there is nothing.

**Comment by Tathanen** [comment]
It'll broadcast `APIRequestCanceled` for each canceled request, per the standard `$cancel` behavior.

**Pull request merged**